Optimize docker image

majkus522 opened the pull request:
Reduce docker image size nearly 3 times.

majkus522 committed:
Optimize docker image

man90es commented on Dockerfile:
```diff
@@ -1,16 +1,23 @@
 RUN npm install -g pnpm
-
 COPY package*.json ./
-COPY pnpm-lock.yaml ./
```
Please keep pnpm-lock.yaml during package installation, it ensures that all package versions are always exactly the same even if package.json allows wiggle room. Sometimes even a patch version can change a small thing that leads to instability

man90es commented on Dockerfile:
```diff
@@ -1,16 +1,23 @@
+WORKDIR /usr/src/app
+RUN npm install -g pnpm
+COPY package*.json ./
+RUN pnpm install --production --quiet && rm -rf ~/.local/share && rm -rf ~/.cache
```
I see you install pnpm and run pnpm install in both stages, that can increase build time significantly, so how about this:
  - Add a stage `base` at the top and install pnpm inside, then use it to create `builder` and `main` stages so that they inherit installed pnpm 
  - Copy `node_modules` from `builder` to `main` and run `pnpm prune --prod` instead of installing again

man90es reviewed (changes requested):
Good PR overall, but some tweaking is needed

majkus522 committed:
Fixed small issues

I also moved character icon generation to separate npm script so that there is no need to copy assets to the builder stage

man90es reviewed (approved):
LGTM
